sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -209,12 +209,20 @@ class SessionCatalog(
           "you cannot create a database with this name.")
     }
     validateName(dbName)
-    val qualifiedPath = makeQualifiedPath(dbDefinition.locationUri)
     externalCatalog.createDatabase(
-      dbDefinition.copy(name = dbName, locationUri = qualifiedPath),
+      dbDefinition.copy(name = dbName, locationUri = makeQualifiedDBPath(dbDefinition.locationUri)),
       ignoreIfExists)
   }
 
+  private def makeQualifiedDBPath(locationUri: URI): URI = {
+    if (locationUri.isAbsolute) {
+      locationUri
+    } else {
+      val fullPath = new Path(conf.warehousePath, CatalogUtils.URIToString(locationUri))
+      makeQualifiedPath(fullPath.toUri)
```

Comment:
do we still need to call `makeQualifiedPath`? I think the `fullPath` is already qualified here.

Reply:
the warehousePath can accept a relative path, the `fullPath` might not be qualified yet

Reply:
shall we make sure it's not relative? I think it's confusing if users set a relative path to the warehouse path.

It's not related to this PR though.

Reply:
Now it is relative to $PWD or Hadoop user home directory. I think such a change would bring serious behavior change, I'm not sure whether worth to do it or not.